src/Compiler/CompilerDialects.cpp

```diff
@@ -0,0 +1,48 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+//===------------------------ CompilerDialects.cpp ------------------------===//
+
+#include "CompilerDialects.hpp"
+
+#include "src/Dialect/Krnl/KrnlOps.hpp"
+#include "src/Dialect/ONNX/ONNXDialect.hpp"
+
+#include "mlir/InitAllDialects.h"
+
+using namespace mlir;
+
+namespace onnx_mlir {
+
+DialectRegistry registerDialects(ArrayRef<accel::Accelerator::Kind> accels) {
+  DialectRegistry registry;
+
+  // Note that we cannot consult command line options because they have not yet
+  // been parsed when registerDialects() is called.
+
+  registry.insert<arith::ArithDialect>();
+  registry.insert<linalg::LinalgDialect>();
+  registry.insert<affine::AffineDialect>();
+  registry.insert<LLVM::LLVMDialect>();
+  registry.insert<scf::SCFDialect>();
+  registry.insert<func::FuncDialect>();
+  registry.insert<vector::VectorDialect>();
+  registry.insert<shape::ShapeDialect>();
+  registry.insert<math::MathDialect>();
+  registry.insert<memref::MemRefDialect>();
+  registry.insert<ONNXDialect>();
+  registry.insert<KrnlDialect>();
+  registry.insert<cf::ControlFlowDialect>();
+
+  // Initialize accelerator(s) if required.
+  accel::initAccelerators(accels);
+
+  // Register dialects for accelerators.
```

Comment:
Thanks for maintaining the separation between normal code and accelerator.

I saw that loading dialects in the NNPA code were removed, so they can just be registered?
@tungld maybe you could check that this works for our usage?

Reply:
> I saw that loading dialects in the NNPA code were removed, so they can just be registered?

yes, getOrLoadDialects() is no longer needed because we now call registerDialects() and register the dialects with the context and then load them with `ctx->loadAllAvailableDialects()`